specification/oracle/models/oraclesubscription/oracleSubscription.tsp

```diff
@@ -0,0 +1,177 @@
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+
+namespace Oracle.Database;
+
+@doc("OracleSubscriptionProvisioningState enum")
+union OracleSubscriptionProvisioningState {
+  ResourceProvisioningState,
+  string,
+}
+
+@singleton
```

Comment:
I am a little unsure if singleton is really what we  want here?  https://azure.github.io/typespec-azure/docs/howtos/ARM/resource-type#singleton-resource

Reply:
Yes that is by design, customer can have only one OracleSubscription in their Azure Subscription

Reply:
That is by design, customer allowed to have only one OracleSubscription per Azure Subscription